util: Pass pthread_self() to pthread_setschedparam instead of 0

laanwj opened the pull request:
Nowhere in the man page of `pthread_setschedparam` it is mentioned that `0` is a valid value. The example uses `pthread_self()`, so should we.

(noticed by Anthony Towns)
Fixes #12915.

laanwj committed:
util: Pass pthread_self() to pthread_setschedparam instead of 0

Nowhere in the man page of `pthread_setschedparam` it is mentioned that
`0` is a valid value. The example uses `pthread_self()`, so should we.

(noticed by Anthony Towns)

MarcoFalke commented:
@eklitzke Mind to take a look here?

MarcoFalke commented:
Concept ACK cff66e6a29 (fixes the issue for me, didn't review though)

practicalswift commented:
utACK cff66e6a29d700188f6ecebeaacc8bcaa1fc1095

ajtowns commented:
utACK cff66e6a29d700188f6ecebeaacc8bcaa1fc1095

Neither clang nor gcc seems to warn about designated initializers with c++11. I think this is the only use of one in the codebase currently, so removing it is probably fine.

laanwj committed:
util: Remove designator initializer from ScheduleBatchPriority

Although no compiler appears to complain about it, these are
not valid for c++11.
(http://en.cppreference.com/w/cpp/language/aggregate_initialization says they're c++20)

The structure is defined as:

   struct sched_param {
       int sched_priority;
   };

So passing 0 for the first field has the same effect.

promag commented:
utACK cff66e6.

> Should I make this change here too?

Maybe just fix the problem for now.

promag commented:
utACK b86730a.

sidhujag commented:
I Noticed this aswell.. fixed it with self()
